mantle/kola/harness.go

```diff
@@ -627,26 +624,7 @@ Environment=KOLA_UNIT=%s
 Environment=%s=%s
 ExecStart=%s
 `, KoletExtTestUnit, kolaExtBinDataEnv, kolaExtBinDataDir, remotepath)
-	runextconfig := ignv3types.Config{
-		Ignition: ignv3types.Ignition{
-			Version: "3.0.0",
-		},
-		Systemd: ignv3types.Systemd{
-			Units: []ignv3types.Unit{
-				{
-					Name:     KoletExtTestUnit,
-					Contents: &unit,
-					Enabled:  util.BoolToPtr(false),
-				},
-			},
-		},
-	}
-
-	finalIgn := ignv3.Merge(ignc3, runextconfig)
-	serializedIgn, err := json.Marshal(finalIgn)
-	if err != nil {
-		return errors.Wrapf(err, "serializing ignition")
-	}
+	config.AddSystemdUnit(KoletExtTestUnit, unit, conf.Enable)
```

Comment:
Hmm, seems to work here. Are you referring to the v2 to v2.1 transition where the field went `Enable string` to `Enabled *string`? Looking in `conf.go` it seems like it's wired correctly.

Reply:
He's saying the unit was `Enabled:  util.BoolToPtr(false),` i.e. not enabled, and it definitely should not be enabled by default or it will introduce race conditions now - the harness needs to set things up.

Reply:
Oh yes, nice catch! I somehow thought you meant `conf.Enable` didn't actually mean enabling the service.